internal/monitors/elasticsearch/client/nodestatsfetcher.go

```diff
@@ -0,0 +1,164 @@
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.mapped.count", dims, jvm.BufferPools.Mapped.Count),
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.mapped.used_in_bytes", dims, jvm.BufferPools.Mapped.UsedInBytes),
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.mapped.total_capacity_in_bytes", dims, jvm.BufferPools.Mapped.TotalCapacityInBytes),
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.direct.count", dims, jvm.BufferPools.Direct.Count),
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.direct.used_in_bytes", dims, jvm.BufferPools.Direct.UsedInBytes),
+			prepareGaugeHelper("gauge.jvm.mem.buffer_pools.direct.total_capacity_in_bytes", dims, jvm.BufferPools.Direct.TotalCapacityInBytes),
+			prepareGaugeHelper("gauge.jvm.classes.current-loaded-count", dims, jvm.Classes.CurrentLoadedCount),
+
+			prepareCumulativeHelper("counter.jvm.gc.count", dims, jvm.JvmGcStats.Collectors.Young.CollectionCount),
+			prepareCumulativeHelper("counter.jvm.gc.old-count", dims, jvm.JvmGcStats.Collectors.Old.CollectionCount),
+			prepareCumulativeHelper("counter.jvm.gc.old-time", dims, jvm.JvmGcStats.Collectors.Old.CollectionTimeInMillis),
+			prepareCumulativeHelper("counter.jvm.classes.total-loaded-count", dims, jvm.Classes.TotalLoadedCount),
+			prepareCumulativeHelper("counter.jvm.classes.total-unloaded-count", dims, jvm.Classes.TotalUnloadedCount),
+		}...)
+	}
+
+	out = append(out, []*datapoint.Datapoint{
+		prepareGaugeHelper("gauge.jvm.mem.heap-used", dims, jvm.JvmMemStats.HeapUsedInBytes),
+		prepareGaugeHelper("gauge.jvm.mem.heap-committed", dims, jvm.JvmMemStats.HeapCommittedInBytes),
+		prepareCumulativeHelper("counter.jvm.uptime", dims, jvm.UptimeInMillis),
+		prepareCumulativeHelper("counter.jvm.gc.time", dims, jvm.JvmGcStats.Collectors.Young.CollectionTimeInMillis),
+	}...)
+
+	return out
+}
+
+func (processStats *Process) fetchProcessStats(enhanced bool, dims map[string]string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+
+	if enhanced {
+		out = append(out, []*datapoint.Datapoint{
+			prepareGaugeHelper("counter.process.max_file_descriptors", dims, processStats.MaxFileDescriptors),
+			prepareGaugeHelper("gauge.process.cpu.percent", dims, processStats.CPU.Percent),
+			prepareCumulativeHelper("counter.process.cpu.time", dims, processStats.CPU.TotalInMillis),
+			prepareCumulativeHelper("counter.process.mem.total-virtual-size", dims, processStats.Mem.TotalVirtualInBytes),
+		}...)
+	}
+
+	out = append(out, []*datapoint.Datapoint{
+		prepareGaugeHelper("gauge.process.open_file_descriptors", dims, processStats.OpenFileDescriptors),
+	}...)
+
+	return out
+}
+
+func fetchThreadPoolStats(enhanced bool, threadPools map[string]ThreadPoolStats, defaultDims map[string]string, selectedThreadPools []string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+	for threadPool, stats := range threadPools {
+		if !utils.StringInSlice(threadPool, selectedThreadPools) {
```

Comment:
Same thing here about using a "set" with a map to bools to make this constant time lookup.

Reply:
Updated this so, the input list is converted into map before using it inside `nodestatsfetcher`